src/codegen/compile.ml

```diff
@@ -5795,23 +5796,94 @@ module FuncDec = struct
       compile_unboxed_const (E.add_fun_ptr env (E.built_in env reject_name)) ^^
       get_cb_index
 
+  let closures_to_self_reply_reject_callbacks env ts =
```

Comment:
This code looks almost identical to the one above? I wonder if its possible to actually share the implementation, if, instead of passing in the array components, you just pass in the get_array code.  Then I think you would also wind up sharing the code for the success continuations (that do deserialization). But maybe I'm missing some essential difference... 
Try to refactor in a branch off this one to see?

Reply:
I agree, this is grossly duplicated. Will look into it how to bring them together.

Reply:
Done. In a ~~few~~ bunch of commits!